### PR TITLE
User login tests

### LINKED ISSUE #97: Add test for legacy code: userLogin

### PR DESCRIPTION
**Pull request by leahriffell**
**What was changed:**
- Added test for legacy userLogin query
- Added sad path test for userLogin (client does not pass in any requested fields back)

closes #97

### CODE REVIEW

File: code/api/src/modules/user/tests/userQuery.test.js [hunk 1]
```diff
@@ -197,4 +197,59 @@ describe('User queries', () => {
...
+  it('attempts to log in user without passing any response requests', async () => {
+    const bad_request = await request(server)
+      .post('/')
+      .send({ query: '{ userLogin { } }' });
+    expect(bad_request.status).to.eq(400);
+    bad_request.body.should.have.property('errors')
+    bad_request.body.errors[0].should.have.property('message')
+    bad_request.body.errors[0].message.should.be.a('string');
+    expect(bad_request.body.errors[0].message).to.eq('Syntax Error: Expected Name, found }');
+  });
```
Comment (priyapower) [inline code/api/src/modules/user/tests/userQuery.test.js]: Woo! Sad path testing!

File: code/api/src/modules/user/tests/userQuery.test.js [hunk 2]
```diff
@@ -197,4 +197,59 @@ describe('User queries', () => {
...
+    expect(user_request.body.data.userLogin.user.zipcode).to.eq(85623)
+
+    user_request.body.data.userLogin.user.should.have.property('description')
+    expect(user_request.body.data.userLogin.user.description).to.eq('Tattooed seitan waistcoat austin asymmetrical chambray hot chicken man bun poke')
+  });
+
+  it('attempts to log in user without passing any response requests', async () => {
+    const bad_request = await request(server)
+      .post('/')
+      .send({ query: '{ userLogin { } }' });
```
Comment (priyapower) [inline code/api/src/modules/user/tests/userQuery.test.js:247]: This sad path is if the user logs in with blank fields

- Should we also add 1. user logs in with email not in system; 2. user logs in with correct email, wrong password?

File: code/api/src/modules/user/tests/userQuery.test.js [hunk 3]
```diff
@@ -197,4 +197,59 @@ describe('User queries', () => {
 
     expect(bad_request.body.data.user).to.eq(null);
   })
+
+  it('logs in a user', async () => {
```
Comment (priyapower) [inline code/api/src/modules/user/tests/userQuery.test.js:201]: This test looks great!

File: code/api/src/modules/user/tests/userQuery.test.js [hunk 4]
```diff
@@ -197,4 +197,59 @@ describe('User queries', () => {
...
+    const user_request = await request(server)
+      .post('/')
+      .send({ query: '{ userLogin(email: "user@crate.com", password: "123456", role: "USER") { user {id name email password role image address_line1 address_line2 city state zipcode description} token } }'});
```
Comment (priyapower) [inline code/api/src/modules/user/tests/userQuery.test.js:204]: I'm glad that login resolver now exposes all this data for the FE. Thanks for ensuring that

File: code/api/src/modules/user/tests/userQuery.test.js [hunk 5]
```diff
@@ -197,4 +197,59 @@ describe('User queries', () => {
...
+    expect(user_request.body.data.userLogin.user.id).to.eq(2)
+
+    user_request.body.data.userLogin.user.should.have.property('name')
+    expect(user_request.body.data.userLogin.user.name).to.eq('The User')
+
+    user_request.body.data.userLogin.user.should.have.property('email')
+    expect(user_request.body.data.userLogin.user.email).to.eq('user@crate.com')
+
+    user_request.body.data.userLogin.user.should.have.property('password')
+    expect(user_request.body.data.userLogin.user.password).to.eq('$2b$10$pTdwV6spenCVU/iga.o36Ox0.8tFyBcHFW59HSffEkN9TVcyE3H62')
```
Comment (priyapower) [inline code/api/src/modules/user/tests/userQuery.test.js:220]: Awesome that the password shows up hashed

**Review by priyapower — APPROVED** [review]
See the comments for suggestions for further sad pathing

File: code/api/src/modules/user/tests/userQuery.test.js (hunk 2 above)
Comment (leahriffell) [reply code/api/src/modules/user/tests/userQuery.test.js:247]: @priyapower  good call!
 - renamed this test name to `attempts to log in user without passing in login information`
 - added sad path tests for incorrect email and/or password

File: code/api/src/modules/user/tests/userQuery.test.js [hunk 6]
```diff
@@ -197,4 +197,81 @@ describe('User queries', () => {
...
+  it('attempts to log in user with incorrect email', async () => {
+    const bad_request = await request(server)
+      .post('/')
+      .send({ query: '{ userLogin(email: "baduser@crate.com", password: "123456", role: "USER") { user {id} token } }'});
+    expect(bad_request.status).to.eq(200);
+    bad_request.body.should.have.property('errors')
+    bad_request.body.errors[0].should.have.property('message')
+    bad_request.body.errors[0].message.should.be.a('string');
+    expect(bad_request.body.errors[0].message).to.eq('We do not have any user registered with baduser@crate.com email address. Please signup.');
+  });
```
Comment (priyapower) [inline code/api/src/modules/user/tests/userQuery.test.js:275]: Thanks for adding in these sad path tests!

**Pull request merged**